addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -408,7 +408,8 @@ var FileWidget = SearchableMediaWidget.extend({
         }
 
         const img = selected[0];
-        if (!img || !img.id) {
+        const isUnchanged = this.$media.attr('src') === img.image_src;
+        if (!img || !img.id || isUnchanged) {
```

Comment:
@sdegueldre I am digging up old PRs. Could you check if still needed and if so, there is something strange here: you have a condition checking if `img` is falsy after having used it as an object in the line just above.

Reply:
Yes, it's still "needed", Although I'm unconvinced that the problem this fixes is very common in practice, or that the defect is very impactful (basically just some duplicated image data in your database when you double click an image to change it, change your mind, and click the already selected image instead of closing the dialog with the cross or cancel button).

I have fixed the issue you mention in your comment.